superset-frontend/cypress-base/cypress/integration/explore/control.test.ts

```diff
@@ -150,6 +150,45 @@ describe('Time range filter', () => {
   });
 });
 
+describe('AdhocFilter control', () => {
```

Comment:
`AdhocFilterControl` already has its own test file: https://github.com/apache/incubator-superset/blob/901a42b1df68423e176705740e0c589727bbafb8/superset-frontend/cypress-base/cypress/integration/explore/AdhocFilters.test.ts#L52

The auto popup behavior was actually captured by the test, but disabled by this line: https://github.com/apache/incubator-superset/commit/901a42b1df68423e176705740e0c589727bbafb8#diff-3988fce84e5b2c937f6d320addc74db15b55519f5234b116657cc8c1b9db0f81R77

You can just remove this line (and probably a comment saying how this will capture the case that adding a new filter will auto-open the popup).

Reply:
Thanks for pointing that out. I removed this test and added checking the option label content after saving to AdhocFilters.test.ts